include/d2tmc.h

```diff
@@ -18,6 +18,34 @@
 
 #ifndef D2TMC_H
 
+#include "cGame.h"
+#include "definitions.h"
+#include "map/cMap.h"
+#include "structs.h"
+#include "utils/cStructureUtils.h"
+
+#include <allegro/palette.h>
+
+class cAbstractStructure;
```

Comment:
this makes my eyes bleed, but this is how it should be then?

Reply:
I know, right? Thing is, this file declares all the globals. If we do not use forward declarations here, then this will be the header that includes everything. 
I wanted to get rid of d2tmh.h first. Then we can tackle this file. That would require some architectural changes, where, for instance, the *Info structs go into a game or session class that keeps all game data. An object of that type could then be created when a game is started. That would also help with save-files, because if that class has all the necessary data, saving is just a serialization of the contents to disk. But I digress. :)

Baby steps, I'm afraid.

Reply:
cool, since I really would like saving at some point. Although atm I was thinking of having a separate data structure for that with all kinds of mapping in between. (to keep savefiles sane)